src/mem/alloc.h

```diff
@@ -196,18 +307,32 @@ namespace snmalloc
     void set_large_size(void* p, size_t size)
     {
       size_t size_bits = bits::next_pow2_bits(size);
-      set(p, static_cast<uint8_t>(size_bits));
-      // Set redirect slide
-      auto ss = address_cast(p) + SUPERSLAB_SIZE;
-      for (size_t i = 0; i < size_bits - SUPERSLAB_BITS; i++)
-      {
-        size_t run = 1ULL << i;
-        PagemapProvider::pagemap().set_range(
-          ss, static_cast<uint8_t>(64 + i + SUPERSLAB_BITS), run);
-        ss = ss + SUPERSLAB_SIZE * run;
-      }
-      PagemapProvider::pagemap().set(
-        address_cast(p), static_cast<uint8_t>(size_bits));
+      if constexpr (SNMALLOC_PAGEMAP_POINTERS)
+      {
+        set(p, pointer_offset(p, size_bits));
+        // Set redirect slide
+        char* ss = reinterpret_cast<char*>(pointer_offset(p, SUPERSLAB_SIZE));
+        for (size_t i = 0; i < size_bits - SUPERSLAB_BITS; i++)
+        {
+          size_t run = 1ULL << i;
+          PagemapProvider::pagemap().set_range(
+            address_cast(ss), pointer_offset(p, 64 + i + SUPERSLAB_BITS), run);
+          ss += SUPERSLAB_SIZE * run;
+        }
+      }
+      else
```

Comment:
The two branches here, seem to be needlessly different
* `ss += ... ` versus `ss = ss + ...` 
* Different type of ss
* Use of address cast in different places. 
I think I would sooner, you push the `if` to be there twice, once around the `set` and once around the `set_range`, then it is clearer what the differences are.